### DISCUSSION TITLE
"The Mirth Connect Service could not be started" - TLS issue

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#5471] [Q&A]
Hi,

After having run with no issues whatsoever since its first installation, this morning all of a sudden I wasn't able to start Mirth Connect Administrator (3.12). The Mirth Connect service appeared to have stopped, and attempts to start it eventually result in the message "The Mirth Connect Service could not be started."

I have started mcserver.exe by double clicking it, which revealed the following:
_The driver could not establish a secure connection to SQL Server by using Secure Sockets Layer encryption. Error: "The server selected protocol version TLS10 is not accepted by client preferences [TLS13, TLS12]"._

These are perhaps relevant settings that I found in mirth.properties:

```
https.client.protocols = TLSv1.3,TLSv1.2,TLSv1.1
https.server.protocols = TLSv1.3,TLSv1.2,TLSv1.1,SSLv2Hello
```

This website https://itechbrand.com/mirth-error-connecting-server mentions that "Older versions of the TLS protocol have been found to restore Mirth’s functionality and allow you to make a valid connection to the server." so that looks like the path to follow, but also warns that earlier versions are extremely insecure and may compromise your machine's security. So I am hestitant to experiment with this - I actually do not want to do that.

How am I to proceed from here?

**Comment by ab-20-m** [comment]
The SQL Server instance Mirth is trying to connect to should be reconfigured to enable TLS 1.2.

**Reply by Coooz** [reply]
Thank you @ab-20-m . Luckily that's not up to me 🙂.

Do you by chance have an idea why this could have happened out of the blue? I mean yesterday the reconfiguration was not necessary apparently. Could some update be the cause of this?
This will help me in explaining why I want SQL Server reconfigured.

**Reply by ab-20-m** [reply]
Probably Java was updated. The [Oracle JRE and JDK Cryptographic Roadmap](https://www.java.com/en/jre-jdk-cryptoroadmap.html) shows when things change like default versions of TLS, disabling of versions of TLS, etc. have happened.

**Reply by Coooz** [reply]
I'll suggest this to the relevant people and let you know whether your suggestion worked.

**Reply by Coooz** [reply]
I needed to reinstall Mirth Connect. A colleague of mine appeared to have broken something accidentally. 
Reinstalling Mirth Connect fixed the issue.
